src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -6270,6 +6273,30 @@ bool Unit::isPossessing() const
         return false;
 }
 
+bool Unit::IsCharmerOrSelfPlayer() const
```

Comment:
isn't this just `GetCharmerOrSelf()->IsPlayer()`?

Reply:
No because the function is `const` and `GetCharmerOrSelf` is not

Reply:
well, then why is `GetCharmerOrSelf` not const...?

Reply:
`GetCharmerOrSelf` calls `this`. Making the function `const` would make this returns const Unit* which is not what I want from the function.

Reply:
`const_cast` it away in `GetCharmerOrSelf`? seems like a perfectly valid use case for `const_cast`

Reply:
Done